src/DataStructures/FixedHashMap.hpp

```diff
@@ -257,14 +257,14 @@ class FixedHashMapIterator {
     ASSERT(entry_ != nullptr, "Invalid FixedHashMapIterator");
     ASSERT(is_set(*entry_),
            "FixedHashMapIterator points to an invalid value in the map.");
-    // deference pointer, then dereference boost::optional
+    // deference pointer, then dereference std::optional
     return **entry_;
   }
   pointer operator->() const noexcept {
     ASSERT(entry_ != nullptr, "Invalid FixedHashMapIterator");
     ASSERT(is_set(*entry_),
            "FixedHashMapIterator points to an invalid value in the map.");
-    return entry_->get_ptr();
+    return &**entry_;
```

Comment:
I understand why, but it sure looks silly. Maybe a code comment for this one too?

Reply:
So here I went with `std::addressof(entry_->value())` Seems better to be expressive in code than using a comment. Is that clear enough? I also change some other places where we do `**` to `t->value()` since I think that's easier to read

Reply:
Great, I agree -- that's far clearer.